buildroot/share/PlatformIO/boards/marlin_Artillery_Ruby.json

```
@@ -0,0 +1,60 @@
+{
+  "build": {
+    "core": "stm32",
+    "cpu": "cortex-m4",
+    "extra_flags": "-DSTM32F401xx",
+    "f_cpu": "84000000L",
+    "hwids": [
+      [
+        "0x0483",
+        "0xDF11"
+      ]
+    ],
+    "mcu": "stm32f401rct6",
+    "variant": "MARLIN_ARTILLERY_RUBY"
+  },
+  "debug": {
+    "jlink_device": "STM32F401RC",
+    "openocd_target": "stm32f4x",
+    "svd_path": "STM32F40x.svd",
+    "tools": {
+      "stlink": {
+        "server": {
+          "arguments": [
+            "-f",
+            "scripts/interface/stlink.cfg",
+            "-c",
+            "transport select hla_swd",
+            "-f",
+            "scripts/target/stm32f4x.cfg",
+            "-c",
+            "reset_config none"
+          ],
+          "executable": "bin/openocd",
+          "package": "tool-openocd"
+        }
+      }
+    }
+  },
+  "frameworks": [
+    "arduino",
+    "stm32cube"
+  ],
+  "name": "STM32F401RC (64k RAM. 256k Flash)",
+  "upload": {
+    "disable_flushing": false,
+    "maximum_ram_size": 65536,
+    "maximum_size": 262144,
+    "protocol": "dfu",
```

I think it's a dfu-util issue. I found something similiar: https://github.com/pine64/pinecil-firmware-updater/issues/6

Try changing the upload protocol from dfu to stlink / jlink to see if it works.

```suggestion
    "protocol": "stlink",
```

---

> Try changing the upload protocol from dfu to stlink / jlink to see if it works.

Bingo! That change is also in line with the other supported boards.

I don't have one of these boards to test on. Can you pull this PR ([or just click this link for the full .zip](https://github.com/thisiskeithb/Marlin/archive/f9e090bbe281962a376da789cf4831c3bb65004a.zip)) and confirm the board boots & works as expected?

---

Ok, I'll test it tomorrow.

---

Sounds good! I’ll get the other Artillery configs ready to merge later today.

---

Confirmed everything working as expected on a Sidewinder X2.